lib/practice_exercises.rb

```diff
@@ -1,13 +1,60 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1) 
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
-end
 
-# Time Complexity: ?
-# Space Complexity: ?
+  # Declare assessing variables
+  string = string.downcase.gsub(/[^0-9a-z]/i, '')
+  yarn = string.chars
+  start = 0
+  finish = yarn.length - 1 
+
+  # Validate 1-letter or empty string arguments
+  if yarn.length == 1 || yarn.length == 0
+    return true
+  end 
+
+  # Account for one-word non-palindrome strings 
+  if yarn[start] != yarn[finish]
+    return false
+  end 
+
+  # Compare interior string values
+  while start < finish
+    if yarn.include?(yarn[start])
+      start += 1
+    elsif yarn.include?(yarn[finish])
+      finish -= 1
+    else
+      if yarn[start] != yarn[finish]
+        return false
+      end 
+      start += 1
+      finish -= 1
+    end 
+  end 
+  true
+end 
+
+# Time Complexity: O(n)
+# Space Complexity: O(n) (Not sure on this; I have yet to find an explanation on Big O that resonates well.)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍